### PR TITLE
Declutter Script Optimization 

### PR DESCRIPTION
**Pull request by kamalbuilds**
## Description

Using generator expression instead of list comprehension inorder to reduce the memory usage.

### Changes

This will create a generator expression that generates the values on the fly and does not store them in a list.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2428"></a>

### CODE REVIEW

File: src/decider/agent/copyrightDeactivationClutterRemovalScript.py [hunk 1]
```diff
@@ -93,13 +93,7 @@ def copyrightDeactivationMain(df, clutter_flag):
...
     full_table_pos = {"Entity": [], "POS_TAG": []}
 
-    for token in doc:
-      if not token.is_punct | token.is_space:
-        pos_dict[token.text] = token.pos_
-
-    for key in pos_dict:
-      full_table_pos["Entity"].append(key)
-      full_table_pos["POS_TAG"].append(pos_dict[key])
+    full_table_pos["Entity"], full_table_pos["POS_TAG"] = zip(*((token.text, token.pos_) for token in doc if not token.is_punct | token.is_space))
```
Comment (Kaushl2208) [inline src/decider/agent/copyrightDeactivationClutterRemovalScript.py:96]: This looks good!
Can you change the doc ==> doc2 ??

I think that is a small bug already there, Ideally it should be the doc2 for which we take POS_TAG and Entity.
@legendarykamal

**Review by Kaushl2208 — CHANGES_REQUESTED** [review]

File: src/decider/agent/copyrightDeactivationClutterRemovalScript.py (hunk 1 above)
Comment (legendarykamal) [reply src/decider/agent/copyrightDeactivationClutterRemovalScript.py]: Thank you for the review , sounds good . Made the change 👍

**Comment by GMishx** [comment]
@legendarykamal Thank you for your contribution. Please follow our [contributing guidelines](https://github.com/fossology/fossology/blob/59c5d1c0b00e06c43a961464596fd702b12f8e95/CONTRIBUTING.md#git-commit-conventions) for the commit conventions.

Also, please squash your commits into 1.

**Review by Kaushl2208 — APPROVED** [review]
Changes looks good @legendarykamal 
Thank you for your contribution. 👍

**Comment by legendarykamal** [comment]
Thank you @Kaushl2208 @GMishx for the feedback and the review , I have rebased my branch 🤝🏻.

**Comment by shaheemazmalmmd** [comment]
As part of GSoC-2023 we are implementing a new script and we will not be using this anymore. So closing this PR. Thank you for your contributions.

**Pull request closed without merging**